samples/99_zil_interpreter/tests/parser_tests_basic.rb

```diff
@@ -0,0 +1,464 @@
+def test_parser_09(args, assert)
+  source = <<-ZIL
+<ROUTINE WITH-LIST (P "TEST" 0)>
+  ZIL
+  parser = Parser.new
+  parser.parse_string(args, source)
+  parsed = parser.expressions[0]
+
+  expected = Syntax::Form.new(
+    :ROUTINE,
+    :"WITH-LIST",
+    Syntax::List.new(:P, "TEST", 0)
+  )
+
+  assert.equal! parsed, expected
+end
+
+# test FIX parsing (all varieties: binary octal and decimal)
+# -----------------------------------------
+def test_parser_10(args, assert)
+  source = <<-ZIL
+<ROUTINE WITH-LIST (#2 0011 *17* 123)>
+  ZIL
+  parser = Parser.new
+  parser.parse_string(args, source)
+  parsed = parser.expressions[0]
+
+  expected = Syntax::Form.new(
+    :ROUTINE,
+    :"WITH-LIST",
+    Syntax::List.new(
+      3, # '#2 0011'
+      15, # '*17*'
+      123) # 123
+  )
+
+  assert.equal! parsed, expected
+end
+
+# test FORM parse with empty FORM element
+# -----------------------------------------
+def test_parser_11(args, assert)
+  source = <<-ZIL
+  <RETURN <>>
+  ZIL
+  parser = Parser.new
+  parser.parse_string(args, source)
+  parsed = parser.expressions[0]
+
+  expected = Syntax::Form.new(
+    :RETURN,
+    Syntax::Form.new)
+
+  assert.equal! parsed, expected
+end
+
+# test FORM parse with empty LIST element
+# -----------------------------------------
+def test_parser_12(args, assert)
+  source = <<-ZIL
+  <RETURN (() ()) ()>
+  ZIL
+  parser = Parser.new
+  parser.parse_string(args, source)
+  parsed = parser.expressions[0]
+
+  expected = Syntax::Form.new(
+    :RETURN,
+    Syntax::List.new(
+      Syntax::List.new,
+      Syntax::List.new),
+    Syntax::List.new)
+
+  assert.equal! parsed, expected
+end
+
+# test multiline string
+# -----------------------------------------
+def test_parser_13(args, assert)
+  source = <<-ZIL
+  <TELL
+"You are standing in an open field west of a white house, with a boarded
+front door.">
+  ZIL
+  parser = Parser.new
+  parser.parse_string(args, source)
+  parsed = parser.expressions[0]
+
+  expected = Syntax::Form.new(
+    :TELL,
+    "You are standing in an open field west of a white house, with a boarded\nfront door.")
+
+  assert.equal! parsed, expected
+end
+
+# test multiline string with escape chars
+# -----------------------------------------
+def test_parser_14(args, assert)
+  # need to escape backslash to get correct result
+  source = <<-ZIL
+  <TELL
+"You are standing in an \\"open\\" field west of a white house, with a boarded
+front door.">
+  ZIL
+  parser = Parser.new
+  parser.parse_string(args, source)
+  parsed = parser.expressions[0]
+
+  expected = Syntax::Form.new(
+    :TELL,
+    "You are standing in an \"open\" field west of a white house, with a boarded\nfront door.")
+
+  assert.equal! parsed, expected
+end
+
+# test DECL for routine
+# -----------------------------------------
+def test_parser_15(args, assert)
+  source = <<-ZIL
+<ROUTINE ROUTINE1 (P1 P2 P3)
+	 #DECL ((P1 P2 P3) FIX)>
+  ZIL
+  parser = Parser.new
+  #parser.log_flag = true
```

Comment:
Still a comment left here ;) Can be deleted I think

Reply:
Fixed.